Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/SendResults.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Concurrent;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+
+namespace Backbone.Modules.Devices.Infrastructure.PushNotifications.DirectPush;
+
+public class SendResult
+{
+    public bool IsSuccess { get; private init; }
+    public bool IsFailure => !IsSuccess;
+    public SendError Error { get; private init; }
+    public DeviceId DeviceId { get; private set; }
+
+    public enum ErrorReason
+    {
+        InvalidHandle,
+        Unexpected
+    }
+
+    public class SendError
+    {
+        public SendError(ErrorReason reason, string message)
+        {
+            Reason = reason;
+            Message = message;
+        }
+
+        public ErrorReason Reason { get; private set; }
+        public string Message { get; private set; }
+    }
+
+    public static SendResult Success(DeviceId deviceId)
+    {
+        return new SendResult { IsSuccess = true, DeviceId = deviceId };
+    }
+
+    public static SendResult Failure(DeviceId deviceId, ErrorReason reason, string message = "")
+    {
+        return new SendResult { IsSuccess = false, Error = new SendError(reason, message), DeviceId = deviceId };
+    }
+}
+
+public class SendResults
```

Comment:
Separate file. And maybe it's time to create a folder for all the SendResult stuff. Then you might also move the inner classes out of `SendResult` to simplify the usage.